### PR TITLE
Unify logging for Play Billing.

### PR DESCRIPTION
**Pull request by PEConn**
This adds debug logging for pretty much everything under the tag `TwaBilling`.

### CODE REVIEW

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/Logging.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package com.google.androidbrowserhelper.playbilling.digitalgoods;
```
Comment (andreban) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/Logging.java:1]: Missing license header.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/Logging.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.google.androidbrowserhelper.playbilling.provider;
```
Comment (andreban) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/Logging.java:1]: Missing license header

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/Logging.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import android.util.Log;
+
+import com.android.billingclient.api.BillingClient;
+import com.android.billingclient.api.BillingResult;
+
+/**
+ * Consolidates all the logging for this package in one place.
+ */
+public class Logging {
+    private static final String TAG = "TwaBilling";
```
Comment (andreban) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/Logging.java:12]: I wonder if it's worth differentiating from the other logging? Eg: `TwaBilling.digitalgoods` and `TwaBilling.provider`.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/Logging.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    static void logGetDetailsResponse(BillingResult result) {
+        int responseCode = result.getResponseCode();
+        Log.d(TAG, "GetDetails returned code " + responseCode);
+        if (responseCode != BillingClient.BillingResponseCode.OK) {
+            Log.d(TAG, result.getDebugMessage());
+        }
+    }
+
+    static void logConnected() {
```
Comment (andreban) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/Logging.java:59]: Are those two intentionally left empty? Can we add a comment saying it if that's the case?

**Review by andreban — CHANGES_REQUESTED** [review]
Looks good. A couple of missing license headers and a couple of questions.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/Logging.java (hunk 4 above)
Comment (PEConn) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/Logging.java:73]: Nope, that was an accident.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/Logging.java (hunk 3 above)
Comment (PEConn) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/Logging.java]: I'd like to just do `adb logcat | grep TwaBilling` and get everything. But yeah, I like your idea, I don't want the tags to get too long so I'll do it as `TwaBilling.P` and `TwaBilling.DG`.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java [hunk 5]
```diff
@@ -31,9 +31,7 @@
 
 public class DigitalGoodsRequestHandler implements ExtraCommandHandler {
     private final BillingWrapper mWrapper;
-    private final BillingWrapper.Listener mListener = result -> {
-        throw new IllegalStateException("DigitalGoods does not support Payment Flows.");
```
[the remark this answers is not in the document]
Comment (PEConn) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:35]: I probably should have made this bit more explicit. But essentially this listener gets triggered by the Play Billing Library whenever someone completes a payment flow. The original reason for having this exception is that the payment flow cannot be triggered through the Digital Goods API. However, apparently the Play Billing Library treats payment flow completion as a global event. This results in the following bug:

1. We use the Digital Goods API to get the price of something (it has its own Play Billing Library connection).
2. We then launch PaymentActivity, which also connects to Play Billing and attaches its own listener.
3. Once the payment flow is complete, both PaymentActivity's listener and DigitalGoodsRequestHandler's listener are triggered.

So basically, if someone used the Digital Goods API before launching the Payment Activity, completing the payment flow would crash the app.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/Logging.java (hunk 1 above)
Comment (PEConn) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/Logging.java:15]: Done

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/Logging.java (hunk 2 above)
Comment (PEConn) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/provider/Logging.java:15]: Done

**Review by andreban — APPROVED** [review]
LGTM, Thanks!
